exercises/practice/robot-name/src/test/java/RobotTest.java

```diff
@@ -34,6 +37,17 @@ public void resetName() {
         assertThat(name).isNotEqualTo(name2);
         assertIsValidName(name2);
     }
+    
+    @Ignore("Remove to run test")
+    @Test
+    public void robotNamesAreUnique() {
+        Set<String> robotNames = new HashSet<>();
+        int sampleSize = 5000;
+        for (int i = 0; i < sampleSize; i++) {
+            robotNames.add(new Robot().getName());
+        }
+        assertThat(robotNames.size()).isEqualTo(sampleSize);
```

Comment:
Missed this the first time, but I'm pretty confident that this works and does what you want more clearly / with a better assertion.

```suggestion
        assertThat(robotNames).hasSize(sampleSize);
```

Reply:
That is better for sure, thanks